src/python/pants/backend/go/util_rules/first_party_pkg.py

```diff
@@ -134,15 +149,65 @@ async def compute_first_party_package_info(
         Get(FirstPartyPkgImportPath, FirstPartyPkgImportPathRequest(request.address)),
         Get(GoModInfo, GoModInfoRequest(owning_go_mod.address)),
     )
+    target = wrapped_target.target
+
+    # Find resource targets.
+    # TODO: This uses ExplicitlyProvidedDependencies to avoid triggering dependency inference which
+    # causes a rule cycle back to this rule as dependency inference tries to analyze dependent
+    # `go_package` targets. There really needs to be a way to limit `DependenciesRequest` to just
+    # targets meeting certain criteria, which would be resource targets in this case.
+    explicit_deps = await Get(
+        ExplicitlyProvidedDependencies, DependenciesRequest(target[Dependencies])
+    )
+    explicit_deps_targets = await Get(Targets, Addresses(explicit_deps.includes))
+    pkg_relative_resource_targets = [
+        tgt
+        for tgt in explicit_deps_targets
+        # Note: target expansion is not ocurring?
+        if tgt.has_field(ResourceSourceField) or tgt.has_field(ResourcesGeneratingSourcesField)
+        # TODO: Currently limited to resources directly under package in source tree. Allow any resource and
+        # have paths be relative to the resources' spec_path.
+        and tgt.address.spec_path.startswith(request.address.spec_path)
```

Comment:
The wildcards with Go file embedding are relative to the package directory and `..` seems to be prohibited (at least by Bazel rules_go). Needs discussion on what we want to support here for `resources` targets not in or under the package directory. For example, what should the embed be relative to in that case?

For now, this code makes a simpler choice and limits resources to be in or under the package directory.

Reply:
> Needs discussion on what we want to support here for resources targets not in or under the package directory. For example, what should the embed be relative to in that case?

That's generally where source roots are applied. It would be a bit awkward to apply them in some cases, but not in others... perhaps something in here should fail-fast if any `resource` dependency is not located below a `go` target in the filesystem?

Reply:
Agreed.